packages/core/src/components/Autocomplete/Autocomplete.jsx

```diff
@@ -237,6 +298,10 @@ Autocomplete.propTypes = {
    * Message users will see when the `loading` prop is passed to `Autocomplete`.
    */
   loadingMessage: PropTypes.node,
+  /**
+   * Do not render the Clear search `<button>` when prop is passed
+   */
+  noClearInput: PropTypes.bool,
```

Comment:
An alternative to prefixing with `no` could be to make it an optional prop that has a defaultValue of true. That works too 👍

Reply:
I like this much better. I changed it to `clearSearchButton` and added a default `true`. Opting out by passing `clearSearchButton={false}` feels more direct and clear.